codalab/lib/cli_util.py

```diff
@@ -116,6 +116,14 @@ def desugar_command(orig_target_spec, command):
 
     def get(dep):  # Return the key
         key, val = parse_key_target(dep)
+        # This check covers two usage errors:
```

Comment:
Seems like this check should happen in the `parse_key_target` function, because there are many places where we need this check.  The error checking around the call sites of `parse_key_target` can probably be cleaned up / unified.

Reply:
Double-checked our code-base and moved the value check to `parse_key_target` function. As for cleaning up the error checking around the call site of `parse_key_target` function, there are three places that use this:
1. [resolve_key_targets](https://github.com/codalab/codalab-worksheets/blob/a7a62e5848083f2f9c948bcd7969d8a8fba535f8/codalab/lib/bundle_cli.py#L578): the following two functions 2 & 3 call `resolve_key_targets` function:
    1. [do_run_command](https://github.com/codalab/codalab-worksheets/blob/master/codalab/lib/bundle_cli.py#L1532): there seems to be no further clean up could be done.
    1. [do_make_command](https://github.com/codalab/codalab-worksheets/blob/master/codalab/lib/bundle_cli.py#L1451): [line 1453](https://github.com/codalab/codalab-worksheets/blob/master/codalab/lib/bundle_cli.py#L1453) comes from when target_spec is an empty string. Then this seems to be a valid check as well.
Not sure if I understand correctly. If I missed anything, please do let me know.